libspeechd still needed when TTS is disabled

### Godot version

d25c3aaaa708b7d8d001f56ca03ffe9bffeb8798

### System information

Arch Linux, GLES 3, Mesa Intel(R) UHD Graphics (CML GT2)

### Issue description

I don't have libspeechd installed on the system. When building Godot, it first shows it's disabling TTS, but a build error still exists:

```
Warning: Speech Dispatcher development libraries not found. Disabling Text-to-Speech support.
Note: Building a debug binary (which will run slowly). Use `target=release_debug` to build an optimized release binary.
Checking for C header file mntent.h... (cached) yes
scons: done reading SConscript files.
scons: Building targets ...
[  4%] Compiling platform/linuxbsd/speechd-so_wrap.c ...
platform/linuxbsd/speechd-so_wrap.c:86:10: fatal error: libspeechd.h: No such file or directory
   86 | #include <libspeechd.h>
      |          ^~~~~~~~~~~~~~
compilation terminated.
```

### Steps to reproduce

Build without libspeechd.

### Minimal reproduction project

_No response_

--- a/platform/linuxbsd/detect.py
+++ b/platform/linuxbsd/detect.py
@@ -329,6 +329,7 @@ def configure(env):
             env.Append(CPPDEFINES=["PULSEAUDIO_ENABLED"])
             env.ParseConfig("pkg-config libpulse --cflags")  # Only cflags, we dlopen the library.
         else:
+            env["pulseaudio"] = False
             print("Warning: PulseAudio development libraries not found. Disabling the PulseAudio audio driver.")
 
     if env["dbus"]:
@@ -343,6 +344,7 @@ def configure(env):
             env.Append(CPPDEFINES=["SPEECHD_ENABLED"])
             env.ParseConfig("pkg-config speech-dispatcher --cflags")  # Only cflags, we dlopen the library.
         else:
+            env["speechd"] = False
             print("Warning: Speech Dispatcher development libraries not found. Disabling Text-to-Speech support.")
 
     if platform.system() == "Linux":
@@ -352,6 +354,7 @@ def configure(env):
                 env.Append(CPPDEFINES=["UDEV_ENABLED"])
                 env.ParseConfig("pkg-config libudev --cflags")  # Only cflags, we dlopen the library.
             else:
+                env["udev"] = False
                 print("Warning: libudev development libraries not found. Disabling controller hotplugging support.")
     else:
         env["udev"] = False  # Linux specific

